From 76d0eccb31de3f0b7ba1124029cae40fca49fea2 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Fri, 25 Jul 2008 23:06:19 +0100
Subject: [PATCH] fix-pcf50633-mask-second-on-resume.patch

We leave SECOND unmasked on resume, it's like the
situation at probe() time, but there it makes us
turn SECOND off after coldplug action.  So we need
to act like after that has happened, not exactly
like what we do at probe / init time.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/i2c/chips/pcf50633.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index 76bb886..352b54a 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -2543,6 +2543,8 @@ static int pcf50633_resume(struct device *dev)
 		dev_err(dev, "Failed to restore LDOs :-( %d\n", ret);
 
 	memset(res, 0, sizeof(res));
+	/* not interested in second on resume */
+	res[0] = PCF50633_INT1_SECOND;
 	ret = i2c_smbus_write_i2c_block_data(&pcf->client,
 					     PCF50633_REG_INT1M,
 					     5, &res[0]);
-- 
1.5.6.3

